### PR TITLE
VACMS-3636 3666 3676: Consolidate multiple clp panel prs.

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
Consolidates multiple clp panel pr's / logic:
https://github.com/department-of-veterans-affairs/va.gov-cms/pull/3636
https://github.com/department-of-veterans-affairs/va.gov-cms/pull/3666
https://github.com/department-of-veterans-affairs/va.gov-cms/pull/3676

## Testing done
Visual / behat

## QA steps
Lot going on here - manual qa by @kevwalsh will be required.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [x] `Product Support Team`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: config/sync/field.field.node.campaign_landing_page.field_clp_connect_with_us.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+third_party_settings:
+  entity_reference_validators:
+    circular_reference: false
+    duplicate_reference: false
+id: node.campaign_landing_page.field_clp_connect_with_us
+field_name: field_clp_connect_with_us
+entity_type: node
+bundle: campaign_landing_page
+label: 'Add a "Connect with us" panel'
+description: 'Choose an administration from this list to display a set of social media and links to email lists'
```
Comment (swirtSJW) [inline config/sync/field.field.node.campaign_landing_page.field_clp_connect_with_us.yml:19]:
```suggestion
description: 'Choose an administration from this list to display a set of social media and links to email lists.'
```

**Review by swirtSJW — COMMENTED** [review]
Overall looks good..  One super minor suggestion and looks like some cleanup is needed for linting failure tests.

**Review by kevwalsh — CHANGES_REQUESTED** [review]
Form validation feedback looks pretty good, although it's weird that it gives a numerical value for one fail, but not others. Not a deal-breaker
Create_Campaign_Landing_Page___VA_gov_CMS_and_Slack___cms-team___Digital_Service___VA___2_new_items_and_CivicActions_-_Calendar_-_Week_of_December_7__2020_and_Create_VAMC_System___VA_gov_CMS

Defect: If i add a node title and click Save without filling out a single field, nothing happens. i assume form validation is somehow choking. No feedback from the UI. 

Video panel defect:  this should have a red asterisk. Form validation doesn't seem to be checking, either. 

Create_Campaign_Landing_Page___VA_gov_CMS


Stories panel defect: this should have a red asterisk. Form validation doesn't seem to be checking, either. 

Create_Campaign_Landing_Page___VA_gov_CMS

Defect: Downloadable resources should have a red asterisk. Form validation doesn't seem to be checking, either. 

Create_Campaign_Landing_Page___VA_gov_CMS

@swirtSJW At some point we may need to decide that form validation issues should be considered as follow-ups. The content model is looking great we want this in Monday's deploy for content entry starting Tuesday.

**Comment by ethanteague** [comment]
@kevwalsh this doesn't match what is building for me locally. I'll inspect.

**Comment by ethanteague** [comment]
@swirtSJW @kevwalsh fixed all items with one exception:  ```Defect: If i add a node title and click Save without filling out a single field, nothing happens. i assume form validation is somehow choking. No feedback from the UI.```
Like you suspected, this is happening b/c html form elements in various panels that have js / behavior validation are empty. The only way I can think of to remedy this is to undo the behaviors logic, and create new constraint handlers for these fields, like I'm doing on paragraphs. Essentially, server side validation to replace browser validation.

**Review by kevwalsh — APPROVED** [review]
QA pass!

**Review by swirtSJW — APPROVED** [review]

**Pull request merged**